src/jit/importer.cpp

```diff
@@ -10056,6 +10070,23 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     }
                 }
 
+                // If this is a local and the local is a ref type, see
+                // if we can improve type information based on the
+                // value being assigned.
+                if (isLocal && (lclTyp == TYP_REF))
+                {
+                    // We should have seen a stloc in our IL prescan.
+                    assert(lvaTable[lclNum].lvArgWrite);
+
+                    const bool isSingleDefLocal =
```

Comment:
Since this has the differences you mentioned vs `lvSingleDef`, maybe this should be something more like `isSingleStlocLocal`?